### ISSUE TITLE
Auto URL detection when creating links

### ISSUE THREAD

**Issue by litera** [daviferreira/medium-editor#304]
It would be great when one selects some text and wants to make it a link that editor would automatically detect whether selection is a URL. It it is, it should be automatically converted to a link
* either without opening popup at all
* opening link creation popup but prepopulating it with the same URL

> This feature should likely get on the list for Medium Editor 2.0

**Comment by NoelBaron** [comment]
I spent a lot of time making this work with MediumJS editor. Used helpful code from Tim Downs to break apart the HTML in a sane manner. Hooked it up to a remote API where the URL was also shortened. Happy to share my approach with @daviferreira, but I'm guessing he's already got an implementation plan in mind.

**Comment by daviferreira** [comment]
Not really, please share :smile:

**Comment by NoelBaron** [comment]
My implementation was sort of crazy. 

1. Support At.JS for twitter-style tagging (@,#,US for user stories, etc)
2. Check for non-anchored urls after space/enter keyup.
3. Enable inline images. 
4. Allow certain website elements (like story cards) to drag into text box as an anchor.

For #2, I had to write and test a fairly comprehensive regex to detect urls that were not already formatted. Then I repurposed a stack overflow post by Tim Downs where he iterated through HTML nodes in order to replace them when certain conditions are met. So once a node is found and a URL detected, the node contents were updated and replaced. Somehow this didn't alter the caret position. 

This approach works fine for small Medium.js contents, but it may be inefficient for larger documents. The code snippet is about 160 lines. @daviferreira could probably optimize the hell out of it ;)
